From 16bc1589de7933b675b4e477a56dbb8cf3250f8d Mon Sep 17 00:00:00 2001
From: Rex Shen <mupingshen@gmail.com>
Date: Mon, 20 Oct 2014 10:49:12 +0800
Subject: [PATCH] if there is no default interface, use any available interface

---
 libc/netbsd/resolv/res_cache.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/libc/netbsd/resolv/res_cache.c b/libc/netbsd/resolv/res_cache.c
index 1a0c40c..7abf894 100644
--- a/libc/netbsd/resolv/res_cache.c
+++ b/libc/netbsd/resolv/res_cache.c
@@ -2615,8 +2615,15 @@ _resolv_populate_res_for_iface(res_state statp)
             return;
         }
         if (if_len == 0) {
-            XLOG("%s: INTERNAL_ERROR: can't find any suitable interfaces.\n", __FUNCTION__);
-            return;
+            // by Rex. if no default interface, use any interface that cached.
+            char *tmp = _find_any_iface_name_locked();
+            if (tmp && (strlen(tmp) < sizeof(statp->iface))) {
+                strncpy(statp->iface, tmp, strlen(tmp));
+                XLOG("%s: no default interface, use %s.\n", __FUNCTION__, statp->iface);
+            } else {
+                XLOG("%s: INTERNAL_ERROR: can't find any suitable interfaces.\n", __FUNCTION__);
+                return;
+            }
         }
     }
 
-- 
1.9.1

